server/graphql/purchase_order/src/purchase_order_queries.rs

```diff
@@ -0,0 +1,155 @@
+use async_graphql::*;
+use graphql_core::{
+    generic_filters::{DatetimeFilterInput, EqualFilterStringInput},
+    map_filter,
+    pagination::PaginationInput,
+    simple_generic_errors::RecordNotFound,
+    standard_graphql_error::StandardGraphqlError,
+    ContextExt,
+};
+use graphql_types::types::{PurchaseOrderConnector, PurchaseOrderNode, PurchaseOrderNodeStatus};
+use repository::{DatetimeFilter, EqualFilter, PaginationOption};
+use repository::{PurchaseOrderFilter, PurchaseOrderSort, PurchaseOrderSortField};
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq)]
+#[graphql(rename_items = "camelCase")]
+pub enum PurchaseOrderSortFieldInput {
+    Supplier,
+    Number,
+    CreatedDatetime,
+    Status,
+    TargetMonths,
+    DeliveryDate,
+    Lines,
+}
+
+#[derive(InputObject)]
+pub struct PurchaseOrderSortInput {
+    /// Sort query result by `key`
+    key: PurchaseOrderSortFieldInput,
+    desc: Option<bool>,
+}
+
+#[derive(InputObject, Clone)]
+pub struct EqualFilterPurchaseOrderStatusInput {
+    pub equal_to: Option<PurchaseOrderNodeStatus>,
+    pub equal_any: Option<Vec<PurchaseOrderNodeStatus>>,
+    pub not_equal_to: Option<PurchaseOrderNodeStatus>,
+}
+
+#[derive(InputObject, Clone)]
+pub struct PurchaseOrderFilterInput {
+    pub id: Option<EqualFilterStringInput>,
+    pub created_datetime: Option<DatetimeFilterInput>,
+    pub status: Option<EqualFilterPurchaseOrderStatusInput>,
+    pub supplier: Option<EqualFilterStringInput>,
+    pub store_id: Option<EqualFilterStringInput>,
+}
+
+#[derive(Union)]
+pub enum PurchaseOrdersResponse {
+    Response(PurchaseOrderConnector),
+}
+
+#[derive(Union)]
+pub enum PurchaseOrderResponse {
+    Error(RecordNotFound),
+    Response(PurchaseOrderNode),
+}
+
+pub fn get_purchase_order(
+    ctx: &Context<'_>,
+    store_id: &str,
+    id: &str,
+) -> Result<PurchaseOrderResponse> {
+    // TODO add auth validation once permissions finalised
+    let service_provider = ctx.service_provider();
+    let service_context = service_provider.context(store_id.to_string(), "".to_string())?;
```

Comment:
Still awaiting auth - so currently passing in empty strings etc

Reply:
Please add this to [To-do](https://github.com/msupply-foundation/open-msupply/issues/8177), or make a sub-issue